docs/source/AdministratorGuide/Resources/identityprovider.rst

```diff
@@ -0,0 +1,59 @@
+.. _resourcesIdProvider:
+
+=================
+Identity Provider
+=================
+
+This type of resource describes the interaction with third-party IdP.
+
+An identity provider (IdP) is a system entity that creates, maintains, and manages identity information for principals and also provides authentication services to relying applications within a federation or distributed network.
+IdP offer user authentication as a service.
+
+These resources are used by the following DIRAC instances:
+
+- Framework/TokenManager service
+- Framework/Auth endpoint
+- web portal
+
+.. note:: With the start of IdP use, the bulk of user management relies on VO managers through the selected IdP.
+
+.. note:: Since not all DIRAC services support token authorization, users still need to keep a fresh proxy in ProxyManager.
+
+The following IdP are presented here: ``OAuth2``, ``CheckIn``, ``IAM``.
+
+------------------------
+OAuth2 identity provider
```

Comment:
Short answer: yes, CheckIn and IAM are inheriting from OAuth2. And yes, OAuth2 can be used stand-alone.

Long answer: 
The `OAuth2IdProvider` class contains a basic interface for interacting with **OAuth2 authorization servers**, `CheckInIdProvider` inheriting from this class by modifying "CheckIn-specific" things. 

```
class CheckInIdProvider(OAuth2IdProvider):
    def getGroupScopes(self, group):
        idPScope = getGroupOption(group, "IdPRole")
        if not idPScope:
            idPScope = "eduperson_entitlement?value=urn:mace:egi.eu:group:%s:role=%s#aai.egi.eu" % (
                getVOForGroup(group),
                group.split("_")[1],
            )
        return S_OK(scope_to_list(idPScope))
```

Now there are some differences between IAM and CheckIn, mainly in the organization of tokens. There are still discussions about using one standard for all, CheckIn has long stated that it will adopt the WLCG standard. So when everyone starts using tokens with the same content, we can get rid of other classes and use only `OAuth2IdProvider`.